### DISCUSSION TITLE
Flexmock object lifetime

### DISCUSSION THREAD

**Discussion by mdwebster** [flexmock/flexmock#169] [Q&A]
I'm sure this is because I'm a python noob but I'm struggling to find the answer.
I have a partial flexmock being set up in a pytest fixture that overrides an open_resource call to the pyvisa library.
This works fine in the test method that uses the fixture the mock is defined in but when the test finishes and the execution goes back to the fixture (after the yield) I can see in the debugger that the method I replaced has reverted to the original method instead of the mocked one. Is there something going on under the hood that tears down the mock when the test function is done?
I wanted to be able to do some final teardown after the yield but I get an expectation exception when calling the mocked function after the yield. Tracing the calls in the debugger I see the expectations are no longer defined at that point. 
This is likely trivial but I would like clarification if anyone knows.
Thank you for your time.

**Comment by ollipa** [comment]
Flexmock resets the mocks automatically when the tests are teared down. When the mocks are teared down, the expectations you set are also checked at the same time. If you want to add more expectations to the mock, you don't need to do it after yield since the expectations are validated lazily.
